shell.nix

```diff
@@ -3,20 +3,15 @@
   target-os ? "all" }:
 
 let
-  projectDeps = import ./default.nix { inherit target-os pkgs nixpkgs-bootstrap; inherit (nixpkgs-bootstrap) config; };
+  project = import ./default.nix { inherit target-os pkgs nixpkgs-bootstrap; inherit (nixpkgs-bootstrap) config; };
+  projectShell = project.shell;
+  shellBootstraper = pkgs.callPackage ./nix/shell-bootstrap.nix { };
   platform = pkgs.callPackage ./nix/platform.nix { inherit target-os; };
-  useFastlanePkg = (platform.targetAndroid && !stdenv.isDarwin);
   # TODO: Try to use stdenv for iOS. The problem is with building iOS as the build is trying to pass parameters to Apple's ld that are meant for GNU's ld (e.g. -dynamiclib)
   stdenv = pkgs.stdenvNoCC;
   mkShell = pkgs.mkShell.override { inherit stdenv; };
-  fastlane = pkgs.callPackage ./fastlane {
-    bundlerEnv = _: pkgs.bundlerEnv { 
-      name = "fastlane-gems";
-      gemdir = ./fastlane;
-    };
-  };
 
-in mkShell {
+in mkShell (shellBootstraper {
```

Comment:
So `mkShell` inherits here things like `buildInputs` passed to `shellBootstrapper` or am I missing something?

Reply:
Yeah, `shellBootstraper` modifies an attribute set and passes that to `mkShell`.